cli/tests/cli_test.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+var (
+	cwd       string
+	cftBinary string
+)
+
+// TestCLI does integration testing.
```

Comment:
I think this function is attempting to be overly generic. We can have a testing function with cases focused solely on testing validation functionality.

If there is useful code which we can share with other integration tests, we could/would put it into shared functions.

Reply:
There has been no existing tests implementation. I've created it being inspired by terraform validator tests.

I haven't moved code that could be shared to separate functions because I've been ignoring other tests completely (as you and @aaron-lane have asked me to do)